File: src/main/kotlin/by/mksn/kwitapi/configuration/ServiceConfiguration.java
package by.mksn.kwitapi.configuration;

import by.mksn.kwitapi.configuration.security.JdbcUserDetailsService;
import by.mksn.kwitapi.repository.UserRepository;
import by.mksn.kwitapi.service.UserService;
import by.mksn.kwitapi.service.impl.UserServiceImpl;
import org.springframework.context.annotation.Bean;

public class ServiceConfiguration {

    @Bean
    public JdbcUserDetailsService userDetailsService(UserService userService) {
        return new JdbcUserDetailsService(userService);
    }

    @Bean
    public UserService userService(UserRepository userRepository) {
        return new UserServiceImpl(userRepository);
    }


}

File: src/main/kotlin/by/mksn/kwitapi/configuration/ServiceConfiguration.kt
package by.mksn.kwitapi.configuration

import by.mksn.kwitapi.repository.*
import by.mksn.kwitapi.service.*
import by.mksn.kwitapi.service.impl.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ServiceConfiguration {

    @Bean
    fun currencyService(currencyRepository: CurrencyRepository): CurrencyService
            = CurrencyServiceImpl(currencyRepository)

    @Bean
    fun walletService(
            walletRepository: WalletRepository,
            transactionRepository: TransactionRepository
    ): WalletService = WalletServiceImpl(walletRepository, transactionRepository)

    @Bean
    fun categoryService(
            categoryRepository: CategoryRepository,
            transactionRepository: TransactionRepository
    ): CategoryService = CategoryServiceImpl(categoryRepository, transactionRepository)

    @Bean
    fun transactionService(transactionRepository: TransactionRepository): TransactionService
            = TransactionServiceImpl(transactionRepository)

    @Bean
    fun remittanceService(
            remittanceRepository: RemittanceRepository,
            walletRepository: WalletRepository
    ): RemittanceService
            = RemittanceServiceImpl(remittanceRepository, walletRepository)

}